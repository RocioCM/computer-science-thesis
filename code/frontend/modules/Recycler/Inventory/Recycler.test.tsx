import TestAppWrapper from '@/common/utils/tests';
import Page from '@/pages/recycler/index';
import { render } from '@testing-library/react';

describe('Recycler - Inventory Page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Matches snapshot', () => {
    const { container } = render(
      <TestAppWrapper>
        <Page />
      </TestAppWrapper>
    );
    expect(container).toMatchSnapshot();
  });
});
