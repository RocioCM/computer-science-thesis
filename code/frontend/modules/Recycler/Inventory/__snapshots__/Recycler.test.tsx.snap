// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recycler - Inventory Page Matches snapshot 1`] = `
<div>
  <main
    class="w-full h-screen p-2xl"
  >
    <header
      class="pb-2xl flex items-center justify-between gap-s"
    >
      <h1>
        Lotes de Material
      </h1>
      <button
        class="relative flex items-center justify-center gap-s rounded-rs font-semibold h-10 disabled:cursor-default disabled:bg-opacity-40  focusButton bg-p1 text-n0 py-s px-m w-max"
        data-testid="button"
      >
        <i
          class="fa-solid fa-plus"
          data-testid="fa-icon"
        />
        Crear lote
      </button>
    </header>
    <div
      class="w-full h-full overflow-auto"
      data-testid="table"
    >
      <table
        class="w-full relative"
      >
        <thead
          class="sticky top-0 bg-n0 rounded-lg text-left"
        >
          <tr>
            <th
              class="h6 p-s last:text-center"
            >
              ID
            </th>
            <th
              class="h6 p-s last:text-center"
            >
              Peso (kg)
            </th>
            <th
              class="h6 p-s last:text-center"
            >
              Tipo de material
            </th>
            <th
              class="h6 p-s last:text-center"
            >
              Fecha de creaci√≥n
            </th>
            <th
              class="h6 p-s last:text-center"
            >
              Estado
            </th>
            <th
              class="h6 p-s last:text-center"
            >
              Acciones
            </th>
          </tr>
        </thead>
        <tbody />
        <tfoot>
          <tr>
            <td
              class="px-4 py-2 text-center"
              colspan="6"
            >
              <span
                class="block shrink-0 animate-spin w-8 text-[2rem] mx-auto mt-8"
                data-testid="loading-spinner"
                style="width: 1.7rem; font-size: 1.7rem;"
              >
                <i
                  class="fa-solid fa-recycle"
                  data-testid="fa-icon"
                />
              </span>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </main>
</div>
`;
